lib/transport/shell.js

```diff
@@ -109,7 +109,7 @@ Shell.prototype.transfer = function(files, remoteDir, options) {
                               (remote.username ? remote.username + '@' : ''),
                               remote.host, remoteDir);
 
-    var command = util.format('rsync --files-from %s %s --rsh="ssh -p%s%s" ./ %s',
+    var command = util.format('cat %s | xargs -n 50 -J % rsync %s --rsh="ssh -p%s%s" % %s',
```

Comment:
1. man xargs: "The current default value for number is 5000"
2. Please use %% (https://nodejs.org/api/util.html#util_util_format_format)
3. Can you verify this?
4. This is a requirement.

I think your changes break the current functionality where I can specify an array of files (in different or same directories) and the whole tree is rebuilt inside the remote's target directory.